command.go

```diff
@@ -230,6 +231,33 @@ func (c *Command) HelpFunc() func(*Command, []string) {
 	}
 }
 
+// Output the usage for the command
+// Used when a user provides invalid input
+// Can be defined by user by overriding UsageFunc
+func (c *Command) Usage() error {
+	c.mergePersistentFlags()
+	err := tmpl(c.getOutOrStderr(), c.UsageTemplate(), c)
```

Comment:
I think this is wrong. It's not using `UsageFunc()` so if someone has set `SetUsageFunc()` it wont print the correct usage.